kybra/compiler/install_rust_dependencies.sh

```diff
@@ -1,27 +1,27 @@
 #!/bin/bash
 
-# TODO If we want to save a little bit of time we might be able to instruct rustup to not install some components initially, like clippy and docs
 # TODO we might want to implement all of this in Python/Rust in the future for platform-independence etc
-# TODO once ic-wasm is no longer required, we can probably just check for the wasm32-wasi target being installed: $global_kybra_rustup_bin target list | grep -q "wasm32-wasi (installed)"
 
 kybra_version="$1"
 rust_version="$2"
 
 global_kybra_config_dir=~/.config/kybra
+global_kybra_version_dir="$global_kybra_config_dir"/"$kybra_version"
 global_kybra_rust_dir="$global_kybra_config_dir"/rust/"$rust_version"
 global_kybra_rust_bin_dir="$global_kybra_rust_dir"/bin
-global_kybra_logs_dir="$global_kybra_rust_dir"/logs
+global_kybra_logs_dir="$global_kybra_version_dir"/logs
 global_kybra_cargo_bin="$global_kybra_rust_bin_dir"/cargo
 global_kybra_rustup_bin="$global_kybra_rust_bin_dir"/rustup
+global_kybra_wasi2ic_bin="$global_kybra_rust_bin_dir"/wasi2ic
+global_kybra_rustc_bin="$global_kybra_rust_bin_dir"/rustc
+global_kybra_candid_extractor_bin="$global_kybra_rust_bin_dir"/candid-extractor
 
 export CARGO_TARGET_DIR="$global_kybra_config_dir"/rust/target
 export CARGO_HOME="$global_kybra_rust_dir"
 export RUSTUP_HOME="$global_kybra_rust_dir"
 
 function run() {
-    ic_wasm_already_installed=$(test -e "$global_kybra_rust_bin_dir"/ic-wasm; echo $?)
-
-    if [ "$ic_wasm_already_installed" -eq 1 ]; then
+    if ! ([ -e "$global_kybra_rustup_bin" ] && [ -e "$global_kybra_cargo_bin" ] && [ -e "$global_kybra_rustc_bin" ] && $global_kybra_rustup_bin target list | grep -q "wasm32-wasi (installed)" && [ -e "$global_kybra_wasi2ic_bin" ] && [ -e "$global_kybra_version_dir"/RustPython ] && [ -e "$global_kybra_candid_extractor_bin" ]); then
```

Comment:
Okay I will just fix it now...it will be nicer and easier to understand and hopefully safer because of that...the saddest part is how long it takes to manually test this for robustness, so I'll have to think about that.

Reply:
This has been fixed and tested manually locally to make sure it can fail and resume at each step